torchaudio/backend/sox_io_backend.py

```diff
@@ -243,12 +246,19 @@ def save(
         format (str, optional):
             Output audio format. This is required when the output audio format cannot be infered from
             ``filepath``, (such as file extension or ``name`` attribute of the given file object).
+        dtype (str, optional)
+            Output tensor dtype.
+            Valid values: ``"uint8", "int16", "int32", "float32", "float64", None``
+            ``dtype=None`` means no conversion is performed.
+            ``dtype`` parameter is only effective for ``float32`` Tensor.
     """
+    if src.dtype == torch.float32 and dtype == None:
```

Comment:
The `save` function has to be compatible with TorchScript compiler. Check out "Type Refinement" section to resolve `None` case.

https://pytorch.org/docs/stable/jit_language_reference.html#optional-type-refinement

I am not sure how TorchScript compiler processes `warnings.warn`. If the torchscript_compatibility test fails, you can move this warnings to `_save` function.